### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by rozbb**
This mostly just copies the CI from `ed25519-dalek`

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -8,94 +8,99 @@ on:
...
-      with:
-        command: test
-        args: --no-default-features --features "std u32_backend"
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - run: rustup target add ${{ matrix.target }}
+      - run: ${{ matrix.deps }}
+      - run: cargo test --target ${{ matrix.target }} --no-default-features --lib --tests
+      - run: cargo test --target ${{ matrix.target }} --no-default-features --features alloc --lib
+      - run: cargo test --target ${{ matrix.target }} --no-default-features --features fast --lib
```
Comment (tarcieri) [inline .github/workflows/rust.yml:32]: The use of --lib and --tests is largely an artifact of which doctests build in different feature combinations. Ideally the rustdoc would be feature gated so it always builds

**Comment by tarcieri** [comment]
This probably needs `**` in the branch name too, right?

**Comment by pinkforest** [comment]
Superceded by: https://github.com/dalek-cryptography/x25519-dalek/pull/109

Added your changes as commit on it.

**Comment by rozbb** [comment]
Superceded by #109

**Pull request closed without merging**